### PR TITLE
add task to create private key and CSR

### PR DESCRIPTION
**Pull request by bengerman13**
## Changes proposed in this pull request:

- Add task to create private key and CSR
- Encrypt sensitive columns
- Add certificate to operation so we can track it through renewal
- Stop using context task, since it reuses the session and that's not what
we wanted

## Security considerations

None

### CODE REVIEW

File: renewer/huey.py [hunk 1]
```diff
@@ -21,9 +21,7 @@
...
 
 # Normal task, no retries
-nonretriable_task = huey.context_task(db.session_handler(), as_argument=True)
+nonretriable_task = huey.task()
 
 # These tasks retry every 10 minutes for four hours.
-retriable_task = huey.context_task(
-    db.session_handler(), as_argument=True, retries=6 * 4, retry_delay=10 * 60
-)
+retriable_task = huey.task(retries=6 * 4, retry_delay=10 * 60)
```
Comment (ccostino) [inline renewer/huey.py:27]: These both look fine, I'm just curious what the difference is between a `context_task` vs. just a `task` - in other words, what was the impetus behind the switch; no more need for the DB session to be passed due to the session handler context managers below?

File: renewer/tasks/letsencrypt.py [hunk 2]
```diff
@@ -59,3 +70,61 @@ def create_user(session, route, operation, acme_user):
...
+
+@huey.retriable_task
+def alb_create_private_key_and_csr(operation_id: int):
+    with session_handler() as session:
+        operation = session.query(DomainOperation).get(operation_id)
+        if operation.certificate is None:
+            # note: we're not linking the cert to the route yet. This is intentional
+            # we're going to link them together once we actually have a certificate
+            # this is to prevent messing with the migrator, until/unless we update it
+            # to understand renewals
```
Comment (ccostino) [inline renewer/tasks/letsencrypt.py:83]: 👍 thanks for including the note!

**Review by ccostino — APPROVED** [review]
This looks good to me, thanks @bengerman13!  Just a couple of notes, but all 👍

File: renewer/huey.py (hunk 1 above)
Comment (bengerman13) [reply renewer/huey.py:27]: a [`context_task`](https://huey.readthedocs.io/en/latest/api.html#Huey.context_task) takes a context-manager object, calls `with ...:` on it, and passes it to the task as the first parameter. 
I think the problem I was running into was that I was using a contextmanager _function_ and not an object, so huey didn't like it.

I could probably convert `session_handler_ to be a class, then pass an object to Huey instead, but I think I'm fine just managing the sessions within the tasks instead for now.

Comment (bengerman13) [reply renewer/huey.py]: I replaced the `session_handler` function with a `SessionHandler` class, so `context_tasks` work now. I also added a tiny blurb on them where we define them

File: renewer/tasks/migrations.py [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
 @huey.huey.periodic_task(crontab(month="*", day="*", hour="6", minute="0"))
 def backport_all_manual_certs():
-    with session_handler() as session:
+    with SessionHandler() as session:
         for instance in find_active_instances(session):
             backport_cert(instance.instance_id)
 
 
-@huey.nonretriable_task
+@huey.retriable_task
```
Comment (ccostino) [inline renewer/tasks/migrations.py:16]: Was this switch intentional?

**Review by ccostino — COMMENTED** [review]
Thanks, @bengerman13!  Everything looks good with the context manager updates, just had one question on a decorator swap with `nonretriable` to `retriable`.

File: renewer/tasks/migrations.py (hunk 3 above)
Comment (bengerman13) [reply renewer/tasks/migrations.py]: nope, and I reverted it

**Review by ccostino — APPROVED** [review]
All good, thanks @bengerman13!

**Pull request merged**